### PR TITLE
Improve typing in layers.surface

### PR DESCRIPTION
**Pull request by dstansby**
# References and relevant issues

# Description

### CODE REVIEW

File: napari/layers/surface/surface.py [hunk 1]
```diff
@@ -441,7 +441,7 @@ def _extent_data(self) -> np.ndarray:
         return extrema
 
     @property
-    def features(self):
+    def features(self) -> pd.DataFrame:
```
Comment (psobolewskiPhD) [inline napari/layers/surface/surface.py:444]: Given the docstring, is there anything that would make mypy happy while preserving the idea that this doesn't only/always return a pandas DF?

Comment (psobolewskiPhD) [reply napari/layers/surface/surface.py:444]: (and also slightly further down)

**Review by psobolewskiPhD — APPROVED** [review]

File: napari/layers/surface/surface.py (hunk 1 above)
Comment (dstansby) [reply napari/layers/surface/surface.py:444]: It does (at the moment) always return a DataFrame, so I think this is the correct annotation (at least for now).

**Comment by Czaki** [comment]
Windows failure is related to things that I try to patch in #6745

**Pull request merged**